src/lib/cache/subCache.cpp

```diff
@@ -1285,3 +1306,33 @@ void subCacheStart(void)
   }
   pthread_detach(tid);
 }
+
+
+
+/* ****************************************************************************
+*
+* subCacheItemErrorStatus - 
+*/
+void subCacheItemErrorStatus(const char* tenant, const char* subscriptionId, int errors)
```

Comment:
Yes... I remember that. There were the days of research on weird crashed that we suspected were in due to cache implementation but and the were (most probably) caused by a bug in the usage of mongo driver and we over-simplified the cache trying to find the bug.

In my opinion that 50% of char* should be "recovered" into std::string to align with the rest of the CB code. In addition, the linked-list structure could be probably be replaced by a std::map (using subscription id) as key to improve search time (O(log n) vs O(n)).

(Not sure if we have an issue about this... in negative case we should create it)